File: youtube-fetch/src/archive-exists.js
import fs from "node:fs/promises";
import path from "node:path";

/** @param {string} archivesDir @param {string} videoId */
export async function archiveExists(archivesDir, videoId) {
  if (!/^[\w-]{11}$/.test(videoId)) return false;
  try {
    await fs.access(path.join(archivesDir, `${videoId}.md`));
    await fs.access(path.join(archivesDir, `${videoId}.json`));
    return true;
  } catch {
    return false;
  }
}

File: youtube-fetch-py/youtube_fetch/archive_exists.py
"""检查归档是否已存在."""

from __future__ import annotations

import re
from pathlib import Path

_ID_RE = re.compile(r"^[\w-]{11}$")


def archive_exists(archives_dir: Path, video_id: str) -> bool:
    if not _ID_RE.match(video_id):
        return False
    return (archives_dir / f"{video_id}.md").is_file() and (archives_dir / f"{video_id}.json").is_file()
